### ISSUE TITLE
`dropTarget` and `stackOffset` issue

### ISSUE THREAD

**Issue by 96LawDawg** [ArnoldSmith86/virtualtabletop#1198]
Let's say you have a holder that has one card in it.  For example, a club.  `stackOffsetX` is set to 40.  If you drop another card in there, everything works fine of course.  But let's say that after the club is in there, you change `dropTarget` to accept only hearts.  When you drop a heart in there, it gets stacked on top of the existing card, completely ignoring the `stackOffsetX`.  It gets even stranger if there are multiple cards in there, some of which meet the new `dropTarget` criteria and some that don't.

**Comment by robartsd** [comment]
Only widgets are only positioned by dropOffset and stackOffset if they match dropTarget. This prevents other widget that may children of the holder from interfering.

**Comment by 96LawDawg** [comment]
Okay, but does it have to be that way?  Can there be an option?  I have a use case for that in my new Pennsylvania Rush game.

**Comment by robartsd** [comment]
I suppose we could create a new property for the behavior; or redefine `alignChildren` such that if it is not simply `true` it specifies which children to align in a way similar to the way `dropTarget` specifies which widgets can be dropped.

**Comment by 96LawDawg** [comment]
Thanks, I was just about to change the label from bug when I saw that you did.  I did change the title.  And this isn't a rush.  I think I have a workaround and in the meantime, the game works fine.  I just wanted an enhancement.  We'll just keep this on the wish list until a bunch of coders fall out of the sky and join VTT.

**Comment by robartsd** [comment]
I started looking at the code to see how my idea of setting `alignChildren` to an object that defines the filter to use on the child widgets and came across the point in the code where I believe changes are needed for #996 (I was having trouble following the code execution path when I was actively looking at the issue months ago).

**Comment by 96LawDawg** [comment]
Is this a good time / place to bring up one other nice-to-have feature for holders?  PCIO let’s you have individual and separate piles within holders. Looks pretty cool. We don’t NEED that but it would make us more compatible with PCIO features and it could be useful in a few games. I never felt like opening an issue for that because it is probably a fair amount of work to implement, but I’ll just throw it out there for now.

**Comment by robartsd** [comment]
Yes, PCIO piles can do some useful stuff, but I'm definitely not planning to attempt to take on anything pile related at this time. What I thought might be an easy fix for #996 is not turning out the way I had hoped.

**Comment by RaphaelAlvez** [comment]
So is this issue about PCIO pikesin holder's or about updating the holder when dropOffset and stackOffest are updated?

**Comment by 96LawDawg** [comment]
Forget PCIO piles in holders.  That was just something that came to mind.  This is about the situation described in the first post.  I haven't had it come up again since then, so it isn't a common issue, even for me.

**Comment by RaphaelAlvez** [comment]
> Forget PCIO piles in holders. That was just something that came to mind. This is about the situation described in the first post. I haven't had it come up again since then, so it isn't a common issue, even for me.

It was 2 lines of code so I had to do it.
